New package: HapSim v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: HapSim
- Repository: https://github.com/lescailab/HapSim.jl
- Created by: @sa-yo-nara
- Version: v0.1.0
- Commit: 88deff37b5e51eb8aaaeb3ce9d43e954803f7905
- Git reference: HEAD

goerz commented:
There seems to be a mismatch between the package and the [README](https://github.com/lescailab/HapSim.jl?tab=readme-ov-file), which appears to describe something called `HAPNEST`. I do not see any documentation for a _Julia_ package `HapSim`.

[At the very least](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4), the README should include a description of the Julia package's purpose and a small usage example in the README. An important part of packages in General is that any potential user can figure out what the package is about and how to get started with using it. That is really difficult when there is no documentation.

In the longer term, I definitely recommend setting up a [Documenter](https://documenter.juliadocs.org/stable/)-based documentation. Before a `v1.0` release, or for smaller packages that can be effectively described entirely with their `README`, that's not a requirement, though.

sa-yo-nara commented:
Hi @goerz 
I've updated the README file as requested. I added references, prerequisites, and a short tutorial with the main commands.
I hope everything looks good now.

Please let me know if there's anything else you'd like me to adjust.

Best regards,
Said

goerz commented:
It's up to you to fix the automerge isues...?

goerz commented:
`Pkg` should (almost) never be a dependency of any package. That's a sign that something is pretty fundamentally wrong. Looking at the repo, the code organization seems to be a mess, so this is probably not anywhere ready to be released as a Julia package. I would recommend that you follow the [standard best practices](https://modernjuliaworkflows.org/writing/#local_packages): You should _only_ have `include` statements in the top [`src/HapSim.jl` file](https://github.com/lescailab/HapSim.jl/blob/master/src/HapSim.jl). You `src` folder should only contain the files listed there, and you should not have `include` statements in any other file (a.k.a., every file must be included only once). Do not use include in your [`test/run_tests.jl` file](https://github.com/lescailab/HapSim.jl/blob/master/test/run_test.jl#L6) to load your package. That should only happen via `using HapSim`. Please also make sure that you run your test suite in [GitHub Actions](https://modernjuliaworkflows.org/sharing/#github_actions)

sa-yo-nara commented:
Hi, I can now see the desired output, but I am confused about the automerge error that appears. The error message says:

RegistryCI.AutoMerge.AutoMergeGuidelinesNotMet("The automerge guidelines were not met.")

It seems that the automerge process failed because some guidelines or conditions required for automatically merging the pull request were not satisfied. However, I don’t fully understand which specific guideline was not met or why this is happening.

JuliaHubRegistrator committed:
New package: HapSim v0.1.0

UUID: 5a22e6b1-9da4-4d8a-9828-7fbb35a05de2
Repo: https://github.com/lescailab/HapSim.jl.git
Tree: 823151890590b3e84443eeb94f4074eeac4bf8c5

Registrator tree SHA: 50f504d641745716a5b3eabaf681d3a4937d2ae3

sa-yo-nara commented:
thanks, i update the project.toml and now all automerge checks are ok
